matching-service/model/redis-repository.js

```diff
@@ -48,3 +48,20 @@ export const deleteMatchInCache = async (difficulty, uuid, socketID) => {
     return false
   }
 }
+
+export const isUserInCache = async (uuid) => {
+  try {
+    // given a single match queue, checks if the user is already in the queue
+    const isUserInQueue = (queue) => queue.some((match) => match.uuid === uuid)
+
+    const difficulties = ['Easy', 'Medium', 'Hard']
+    const cachedQueues = await Promise.all(
+      difficulties.map((d) => client.get(d))
```

Comment:
Didn't use the following form (which would usually work) cos the Redis client has some private fields that don't allow direct access:
```suggestion
      difficulties.map(client.get)
```